src/fauxtoshop.cpp

```diff
@@ -1,517 +1,487 @@
+                col = stringToInteger(inputVector[1]);
+                if (row > 0 && col > 0){
+                    break;
+                }
+            }
+        }
+        cout << "Invalid input! Please try again" << endl;
+    }
+}
+
+/*
+ * Function: calculateRBGColourDifference
+ * Usage: Calculates the difference between 2 colours for edge detection and green screen effects
+ * Function:
+ * a. Separate the pixels into their RGB components
+ * b. Take difference between each pair of RGB values (e.g. 2 reds, 2 blues, 2 greens)
+ * c. Take absolute value between for each difference (use nteger abs function in <math> standard)
+ * d. Take a max of the three differences (max function for 2 values at a time)
+ * scatter radius to provide colour for the pixel in new Grid. Select again, if the random selection
+ * is out of bounds of orignal Grid
+ * Hint: use int data type
+ * Extension: Alternate calcalatons
+ * Params: pixel1_row - integer, pixel1_col - integer, pixel2_row - integer, pixel2_col - integer
+ * ------------------------------------------------------------------------------------------------
+ * Returns: maxDifference - integer
+*/
+int calculateRBGColourDifference(int pixel1, int pixel2){
+    // Instatiate variables for red, green, blue for pixel 1
+    int red1, green1, blue1;
+    // Instatiate variables for red, green, blue for pixel 2
+    int red2, green2, blue2;
+    // Get the individual red green blue elements for both pixel 1 and 2 so they can be compared
+    GBufferedImage::getRedGreenBlue(pixel1, red1, green1, blue1);
+    GBufferedImage::getRedGreenBlue(pixel2, red2, green2, blue2);
+    // Calculate the difference between the red, green, blue components and take the absolute
+    // (positive) value
+    int redDiff = abs(red1 - red2);
+    int greenDiff = abs(green1 - green2);
+    int blueDiff = abs(blue1 - blue2);
+    int maxDifference = max(redDiff, greenDiff);
+    maxDifference = max(maxDifference, blueDiff);
+    return maxDifference;
+}
+
+/*
+ * Function: getRandomPixelColor
+ * Usage: For a specific pixel, gets the colour of a nearby pixel within the scatter radius
+ * Function:
+ * a. Calculate the random row and random column value for the random pixel based on the
+ * scatter radius
+ * b. Calculate whether the random pixel location is in bounds of the original image grid
+ * c. If in bounds, assign the colour of the random pixel to the specified pixel
+ * Params: original - grid of integers related to the original image,
+ * newImage - grid of integers related to the new image,
+ * row - integer, col - integer, scatterRadius - integer
+ * ------------------------------------------------------------------------------------------------
+ * Returns: None. Void function.
+*/
+void getRandomPixelColor(Grid<int> &original
+                         , Grid<int> &scatterGrid
+                         , int row
+                         , int col
+                         , int scatterRadius){
```

Comment:
Try to keep the commas on the same line as the word that comes before it. E.g.
`Grid<int> &original, Grid<int> &scatterGrid , int row, int col, int scatterRadius`
If you need to list the parameters on separate lines, you can do it like:
```
void getRandomPixelColor(Grid<int> &original,
                         Grid<int> &scatterGrid,
                         int row,
                         int col,
                         int scatterRadius) {
```